server/src/main/java/org/elasticsearch/repositories/Repository.java

```diff
@@ -133,8 +134,9 @@ default Repository create(RepositoryMetaData metaData, Function<String, Reposito
      * @param includeGlobalState include cluster global state
      * @return snapshot description
      */
-    SnapshotInfo finalizeSnapshot(SnapshotId snapshotId, List<IndexId> indices, long startTime, String failure, int totalShards,
-                                  List<SnapshotShardFailure> shardFailures, long repositoryStateId, boolean includeGlobalState);
+    SnapshotInfo finalizeSnapshot(SnapshotId snapshotId, String repository, List<IndexId> indices, long startTime, String failure,
```

Comment:
We shouldn't have to pass the repository name to `finalize`, the repository must know its own name.